packages/test/test-end-to-end-tests/src/test/gc/gcContainerRuntimeCompat.spec.ts

```diff
@@ -44,18 +48,38 @@ describeCompat.skip("GC summary compatibility tests", "FullCompat", (getTestObje
 			},
 			gcOptions: { gcAllowed: true },
 		};
-		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore({
-			defaultFactory: dataObjectFactory,
-			registryEntries: [[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
-			runtimeOptions,
-		});
+		const runtimeFactory = createContainerRuntimeFactoryWithDefaultDataStore(
+			ContainerRuntimeFactoryWithDefaultDataStore,
+			{
+				defaultFactory: dataObjectFactory,
+				registryEntries: [[dataObjectFactory.type, Promise.resolve(dataObjectFactory)]],
+				runtimeOptions,
+			},
+		);
 		return provider.createContainer(runtimeFactory);
 	}
 
-	beforeEach("setupContainer", async () => {
+	beforeEach("ensure old versions are installed", async function () {
+		this.timeout(20_000);
+		await Promise.all(
+			oldVersionNumbers.map(async (version) =>
+				ensurePackageInstalled(pkgVersion, version, false),
+			),
+		);
+	});
+
+	beforeEach("setupContainer", async function () {
 		provider = getTestObjectProvider({ syncSummarizer: true });
 		mainContainer = await createContainer();
-		dataStoreA = (await mainContainer.getEntryPoint()) as ITestFluidObject;
+		if (mainContainer.getEntryPoint !== undefined) {
+			dataStoreA = (await mainContainer.getEntryPoint()) as ITestFluidObject;
+		} else {
+			// Back-compat: versions of container-loader before 2.0.0-internal.3.3.0 don't have a getEntryPoint API.
```

Comment:
@markfields this choice seem reasonable to you?

See also the changes to the test to force non-incremental GC. You suggested this might be the right way to fix things on a teams thread a while back (I'm finally getting back to this task :))

Reply:
ehh actually I want to just clean up how this whole file operates on second thought. It should be leveraging cross-version compat testing infra we've added rather than doing its own versioning math. Ignore for now :)

Reply:
I pushed an update which resolves this. Using this test as a case study, I'm kind of convinced we probably want variants on `describeCompat` which *only* do cross-version testing, and explicitly empower the test author to choose which containers they're making are made with each of the two (or more, but 2 for now) versions involved.

In the meantime, I went with minimal change to make this test reasonably correct. Seems a bit better than having it disabled, though I think it would be more value if the cross-compat matrix we were using for the test actually included versions that weren't on different values of gcFeature (since that requires weakening the assert that the unreferenced timestamp hasn't changed).